### PR TITLE
[SPARK-23174][BUILD][PYTHON] python code style checker update

### PR DESCRIPTION
**Pull request by rekhajoshm**
## What changes were proposed in this pull request?
Referencing latest python code style checking from PyPi/pycodestyle
Removed pending TODO
For now, in tox.ini excluded the additional style error discovered on existing python due to latest style checker (will fallback on review comment to finalize exclusion or fix py)
Any further code styling requirement needs to be part of pycodestyle, not in SPARK.

## How was this patch tested?
./dev/run-tests

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
This doesn't actually run the Python lint in Jenkins build though because there's no change on the files with `.py`. Mind double checking and fixing:

```diff
diff --git a/dev/run-tests.py b/dev/run-tests.py
index 7e6f7ff0603..6fac324972e 100755
--- a/dev/run-tests.py
+++ b/dev/run-tests.py
@@ -576,7 +576,10 @@ def main():
                                 for f in changed_files):
         # run_java_style_checks()
         pass
-    if not changed_files or any(f.endswith(".py") for f in changed_files):
+    if not changed_files or any(f.endswith("lint-python")
+                                or f.endswith("tox.ini")
+                                or f.endswith(".py")
+                                for f in changed_files):
         run_python_style_checks()
     if not changed_files or any(f.endswith(".R") for f in changed_files):
         run_sparkr_style_checks()
```

too?

File: dev/lint-python [hunk 1]
```diff
@@ -35,11 +35,9 @@ compile_status="${PIPESTATUS[0]}"
 
 # Get pep8 at runtime so that we don't rely on it being installed on the build server.
 #+ See: https://github.com/apache/spark/pull/1744#issuecomment-50982162
-#+ TODOs:
-#+  - Download pep8 from PyPI. It's more "official".
-PEP8_VERSION="1.7.0"
+PEP8_VERSION="2.3.1"
 PEP8_SCRIPT_PATH="$SPARK_ROOT_DIR/dev/pep8-$PEP8_VERSION.py"
-PEP8_SCRIPT_REMOTE_PATH="https://raw.githubusercontent.com/jcrocholl/pep8/$PEP8_VERSION/pep8.py"
+PEP8_SCRIPT_REMOTE_PATH="https://raw.githubusercontent.com/PyCQA/pycodestyle/$PEP8_VERSION/pycodestyle.py"
```
Comment (HyukjinKwon) [inline dev/lint-python]: Shall we leave a note that `pep8` is formally renamed to `pycodestyle` to reduce confusion, and in the PR description too?

File: dev/tox.ini [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-[pep8]
-ignore=E402,E731,E241,W503,E226
+[pycodestyle]
+ignore=E402,E731,E241,W503,E226,E722,E741,E305
```
[the remark this answers is not in the document]
Comment (rekhajoshm) [reply dev/tox.ini]: These(E722, E741, E305) are new code style standards added that Spark seems to be violating.For now, added them in ignored.I have raised the doctest bank line as an issue under pep8, as it does not belong within Spark.

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply dev/tox.ini]: Ah, I asked this to double check if this PR fixes the issue linked JIRA. I think we could make another JIRA if this PR doesn't target to fix the linked JIRA. For the best, it'd be nicer if you'd leave a issue link you raised under pep8, if I understood you correctly.

[the remark this answers is not in the document]
Comment (rekhajoshm) [reply dev/tox.ini]: Here - https://github.com/PyCQA/pycodestyle/issues/723

**Review by HyukjinKwon — COMMENTED** [review]
LGTM otherwise.

File: dev/tox.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (rekhajoshm) [reply dev/tox.ini]: Updated SPARK-11222. Added https://issues.apache.org/jira/browse/SPARK-23174.

**Review by ueshin — COMMENTED** [review]
LGTM except one question.

File: dev/lint-python [hunk 3]
```diff
@@ -35,11 +35,10 @@ compile_status="${PIPESTATUS[0]}"
 
 # Get pep8 at runtime so that we don't rely on it being installed on the build server.
 #+ See: https://github.com/apache/spark/pull/1744#issuecomment-50982162
-#+ TODOs:
-#+  - Download pep8 from PyPI. It's more "official".
-PEP8_VERSION="1.7.0"
+# Updated to latest official version for pep8.pep8 is formally renamed to pycodestyle.
```
Comment (HyukjinKwon) [inline dev/lint-python]: nit .. : `pep8.pep8` -> `pep8. pep8`

**Comment by srowen** [comment]
(If this also resolves SPARK-11222 you can also include that in the title.)

**Comment by HyukjinKwon** [comment]
Ah, seems this one doesn't resolve SPARK-11222 too. I think that JIRA describes to add `pydocstyle` check in our project too given my reading of @rekhajoshm's issue link. Will double check and take an action for SPARK-11222 soon. 

This JIRA, SPARK-23174, seems describing simply upgrading the pep8 (formally renamed to `pycodestyle`).

**Comment by HyukjinKwon** [comment]
@rekhajoshm, mind updating variable names and one nit above? Will merge this one as soon as they are fixed.

**Comment by rekhajoshm** [comment]
@HyukjinKwon @ueshin was bit hesitant to change variable/references as pep8 is very clear within python circles as python style checker.however after pondering, have done the needful. please check.thanks

@srowen for SPARK-11222, I had raised a issue under pycodestyle as it does not belong within Spark.

**Comment by HyukjinKwon** [comment]
retest this please

File: dev/tox.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (felixcheung) [reply dev/tox.ini]: should we fix the code that violates

E722: "do not use bare except'"
E741: "ambiguous variable name 'l'",
E305: "expected 2 blank lines after class or function definition"
?

Or that's another PR/JIRA?

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply dev/tox.ini]: Yea .. there look actually many instances about it. I manually ran it against this PR. Will maybe take this one and open a PR separately after double checking each rule. Didn't take a look carefully yet.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**